br/pkg/lightning/backend/external/writer.go

```diff
@@ -488,6 +488,9 @@ func (w *Writer) createStorageWriter(ctx context.Context) (
 
 // EngineWriter implements backend.EngineWriter interface.
 type EngineWriter struct {
+	// Only 1 writer is used for some kv group(data or some index), no matter
+	// how many routines are encoding data, so need to sync write to it.
+	sync.Mutex
```

Comment:
Why here add new Mutex，Could we reuse，global sort mutex that wenjun introduced in https://github.com/pingcap/tidb/pull/47131

Reply:
see comments, i think that's quite clear

> global sort mutex that wenjun introduced in https://github.com/pingcap/tidb/pull/47131

that's shared among all writers